table_helper.cc

```diff
@@ -165,9 +167,18 @@ future<> table_helper::setup_keyspace(cql3::query_processor& qp, service::migrat
         auto ts = group0_guard.write_timestamp();
 
         if (!db.has_keyspace(keyspace_name)) {
+            std::map<sstring, sstring> opts;
+            if (replication_strategy_name == "org.apache.cassandra.locator.NetworkTopologyStrategy") {
+                for (const auto &dc: qp.proxy().get_token_metadata_ptr()->get_topology().get_datacenters())
+                    opts[dc] = replication_factor;
+            }
+            else {
+                opts["replication_factor"] = replication_factor;
+            }
+            auto ksm = keyspace_metadata::new_keyspace(keyspace_name, replication_strategy_name, std::move(opts), std::nullopt, true);
             try {
                 co_await mm.announce(service::prepare_new_keyspace_announcement(db.real_database(), ksm, ts),
-                        std::move(group0_guard), seastar::format("table_helper: create {} keyspace", keyspace_name));
+                                     std::move(group0_guard), seastar::format("table_helper: create {} keyspace", keyspace_name));
```

Comment:
version w/o white-spaces is more readable because the line is long, but that's just a nitpick, no need to fix